src/Symfony/Bridge/PhpUnit/Tests/DeprecationErrorHandler/DeprecationTest.php

```diff
@@ -49,9 +50,110 @@ public function testItRulesOutFilesOutsideVendorsAsIndirect()
         $this->assertFalse($deprecation->isIndirect());
     }
 
+    public function providerIsSelf(): array
+    {
+        return [
+            [true, '', 'MyClass1', ''],
+            [false, '', 'MyClass1', 'dummy_vendor_path'],
+            [
+                true,
+                serialize(['class' => '', 'method' => '', 'deprecation' => '', 'files_stack' => []]),
+                SymfonyTestsListenerForV5::class,
+                '',
+            ],
+            [
+                true,
+                serialize([
+                    'class' => '',
+                    'method' => '',
+                    'deprecation' => '',
+                    'triggering_file' => '',
+                    'files_stack' => [],
+                ]),
+                SymfonyTestsListenerForV5::class,
+                '',
+            ],
+            [
+                false,
+                serialize([
+                    'class' => '',
+                    'method' => '',
+                    'deprecation' => '',
+                    'triggering_file' => 'dummy_vendor_path',
+                    'files_stack' => [],
+                ]),
+                SymfonyTestsListenerForV5::class,
+                '',
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider providerIsSelf
+     */
+    public function testIsSelf(bool $expectedIsSelf, string $message, string $traceClass, string $file): void
+    {
+        $trace = [
+            ['class' => 'MyClass1', 'function' => 'myMethod'],
+            ['class' => $traceClass, 'function' => 'myMethod'],
+        ];
+        $deprecation = new Deprecation($message, $trace, $file);
+        $this->assertEquals($expectedIsSelf, $deprecation->isSelf());
+    }
+
+    public function providerIsIndirectUsesRightTrace(): array
+    {
+        return [
+            [false, '', [['function' => 'myfunc1'], ['function' => 'myfunc2']]],
+            [
+                true,
+                '',
+                [
+                    ['function' => 'myfunc1', 'file' => __DIR__.'/../../../../../../vendor/composer/ClassLoader.php'],
```

Comment:
This must be why the build fails… it looks like you are trying to make it work with vendors from the root dir, instead of with vendors from the component directory, and I think it should be working in both cases. Maybe try using https://www.php.net/manual/en/function.getcwd.php in there?

Reply:
Fixed, thanks